draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1444,10 +1444,29 @@ user agent MUST process the cookie as follows:
     attribute-value (i.e. either "Strict", "Lax", or "None"). Otherwise, set the
     cookie's same-site-flag to "None".
 
-14. If the cookie's `same-site-flag` is not "None", and the cookie is being set
-    from a context whose "site for cookies" is not an exact match for
-    request-uri's host's registered domain, then abort these steps and ignore
-    the newly created cookie entirely.
+14. If the cookie's `same-site-flag` is not "None":
+
+    1.  If the cookie was received from a "non-HTTP" API, and the API was called
+        from a context whose "site for cookies" is not an exact match for
+        request-uri's host's regisgered domain, then abort these steps and
+        ignore the newly created cookie entirely.
```

Comment:
Does this mean that in a context that's not same-site-strict but same-site-lax, a same-site=lax cookie will not be settable by JavaScript? I am not sure such difference between JS and HTTP is desirable.

Hmm. I may be confused here between context's "site for cookies" and "request"'s client's "site for cookies" as used in section 5.2.

(Also a typo in line 1451, regisgered)

Reply:
It's hard to be precise here, because this document doesn't have much relation to HTML. But, my goal is something like this:

When the `document.cookie` setter is executed, execute the "site for cookies" algorithm from https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-03#section-5.2.1. That'll give you either the top-level document's origin's host's registrable domain, or an empty string.

In the case we're debating, it'll give you the registrable domain of the document that was just navigated-to. That is, both `samesite=strict` and `samesite=lax` will be settable.